src/github.com/getlantern/flashlight/client/balancer.go

```diff
@@ -7,40 +7,70 @@ import (
 	"github.com/getlantern/fronted"
 )
 
+// getBalancer waits for a message from client.balCh to arrive and then it
+// writes it back to client.balCh before returning it as a value. This way we
+// always have a balancer at client.balCh and, if we don't have one, it would
+// block until one arrives.
 func (client *Client) getBalancer() *balancer.Balancer {
+
+	// This balChMu will protect balCh and ensure it always have, at most, one
+	// element enqueued.
```

Comment:
Because we're declaring balCh as `client.balCh = make(chan *balancer.Balancer, 1)`, so if we attempt to write more that one messages to that channel it would block.